hspec-discover/test/Test/Hspec/Discover/RunSpec.hs

```diff
@@ -68,7 +68,7 @@ spec = do
           "{-# LINE 1 \"test/Spec.hs\" #-}"
         , "{-# LANGUAGE NoImplicitPrelude #-}"
         , "{-# OPTIONS_GHC -fno-warn-warnings-deprecations #-}"
-        , "module Main where"
+        , "module Main (main, spec) where"
```

Comment:
We don’t have test coverage for the configNoMain=True case. Can you add a test case for `--no-main`?

Reply:
I've added a test.